src/crush/CrushWrapper.cc

```diff
@@ -2351,6 +2351,7 @@ int CrushWrapper::add_simple_rule_at(
   int ret = crush_add_rule(crush, rule, rno);
   if(ret < 0) {
     *err << "failed to add rule " << rno << " because " << cpp_strerror(ret);
+    free(rule);
```

Comment:
there are three places where we fail to free `rule` in the error handling path. could you please fix them all?

Reply:
Noticed, as well as TestOSDMap, I am still working on the rest. Thanks for quick feedback~